packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -1258,17 +1272,20 @@ export class ModularEditBuilder extends EditBuilder<ModularChangeset> {
 		if (length === 0) {
 			return { type: "global" };
 		}
-		const builds: ChangeAtomIdMap<TreeChunk> = new Map();
-		const innerMap = new Map();
-		builds.set(undefined, innerMap);
-		let id = firstId;
-
-		// TODO:YA6307 adopt more efficient representation, likely based on contiguous runs of IDs
-		for (const cursor of content) {
-			assert(!innerMap.has(id), 0x854 /* Unexpected duplicate build ID */);
-			innerMap.set(id, chunkTree(cursor as ITreeCursorSynchronous, defaultChunkPolicy));
-			id = brand((id as number) + 1);
-		}
+		const builds: ChangeAtomIdRangeMap<readonly TreeChunk[]> = new Map();
+		const rangeMap: RangeMap<readonly TreeChunk[]> = [];
+		builds.set(undefined, rangeMap);
+
+		const trees = content.map((cursor) =>
+			chunkTree(cursor as ITreeCursorSynchronous, defaultChunkPolicy),
+		);
```

Comment:
This is mapping the many nodes in `newContent` into many tree chunks.
Instead, we want to create a single tree chunk for all the nodes in `newContent`.

To do that, we can convert the many `newContent` cursors (which are in node mode) into an array of MapTree_s.
Then get a single (field mode) cursor from that using `cursorForMapTreeField`
Then do `const chunk = chunkFieldSingle(fieldCursor, defaultChunkPolicy);`